plugins/jvm-example/src/main/java/org/elasticsearch/plugin/example/ExamplePluginConfiguration.java

```diff
@@ -19,51 +19,35 @@
 
 package org.elasticsearch.plugin.example;
 
-import org.elasticsearch.ElasticsearchParseException;
 import org.elasticsearch.common.inject.Inject;
-import org.elasticsearch.common.xcontent.XContentParser;
-import org.elasticsearch.common.xcontent.yaml.YamlXContent;
+import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.env.Environment;
 
 import java.io.IOException;
 import java.nio.file.Path;
 
-import static java.nio.charset.StandardCharsets.UTF_8;
-import static java.nio.file.Files.newBufferedReader;
-import static org.elasticsearch.common.io.Streams.copyToString;
-
 /**
  * Example configuration.
  */
 public class ExamplePluginConfiguration {
-    private String test = "not set in config";
-
+    
+    private static final String TEST_KEY = "test";
+    
+    private static final String NO_SETTING_FOUND = "not set in config";
```

Comment:
I honestly think constants like this are more confusing than putting the string in the code. I see this and I can't tell what it is for until I scroll to its usage. The `Setting` class does a good job of making it obvious what the parameters do because we know the order.